Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/SettingsBackgroundPriorityFragment.kt

```diff
@@ -40,19 +39,12 @@ class SettingsBackgroundPriorityFragment :
     }
 
     private fun setButtonOnClickListener() {
-        val switch = binding.settingsSwitchRowBackgroundPriority.settingsSwitchRowSwitch
-        val switchRow = binding.settingsSwitchRowBackgroundPriority.settingsSwitchRow
+        val settingsRow = binding.settingsRowBackgroundPriority
 
         // enable background priority
-        setOf(switch, switchRow).forEach {
-            it.setOnClickListener {
-                val isPriorityEnabled = settingsViewModel.isBackgroundPriorityEnabled.value == true
-
-                if (!isPriorityEnabled) {
-                    (requireActivity() as MainActivity).apply {
-                        startActivitySafely(powerManagement.disableBatteryOptimizationsIntent)
-                    }
-                }
+        settingsRow.setOnClickListener {
+            (requireActivity() as MainActivity).apply {
```

Comment:
Could (requireActivity() as MainActivity) to be null?  Add a"?." before the apply.

Reply:
it cant be null but could raise an exception if no main activity is found, i could replace it with a null check that does nothing instead. Which behaviour would you prefer?